File: src/main/java/com/senderman/TgUser.java
package com.senderman;

import org.telegram.telegrambots.meta.api.objects.User;

public class TgUser {

    private final int id;
    protected String name;

    public TgUser(int id, String name) {
        this.id = id;
        this.name = getSafeName(name);
    }

    public TgUser(User user) {
        this.id = user.getId();
        this.name = getSafeName(user.getFirstName());
    }

    private String getSafeName(String name) {
        return name.replace("<", "&lt;").replace(">", "&gt;");
    }

    public int getId() {
        return id;
    }

    public String getName() {
        return name;
    }

    public String getLink() {
        return String.format("<a href=\"tg://user?id=%1$d\">%2$s</a>", id, name);
    }
}

File: src/main/java/com/senderman/TgUser.kt
package com.senderman

import org.telegram.telegrambots.meta.api.objects.User

open class TgUser {

    val id: Int
    val name: String
    val link: String
        get() = "<a href=\"tg://user?id=$id\">$name</a>"

    constructor(id: Int, name: String) {
        this.id = id
        this.name = getSafeName(name)
    }

    constructor(user: User) {
        this.id = user.id
        this.name = getSafeName(user.firstName)
    }

    private fun getSafeName(name: String): String {
        if (name.trim().isBlank()) return "Без имени"
        return name
            .replace("<", "&lt;")
            .replace(">", "&gt;")
            .replace("&", "&amp;")
    }
}